dot/parachain/collator-protocol/message.go

```diff
@@ -175,32 +177,118 @@ const (
 	Seconded
 )
 
+// getDeclareSignaturePayload gives the payload that should be signed and included in a Declare message.
+// The payload is a the local peed id of the node, which serves to prove that it controls the
+// collator key it is declaring and intends to collate under.
+func getDeclareSignaturePayload(peerID peer.ID) []byte {
+	payload := []byte("COLL")
+	payload = append(payload, peerID...)
+
+	return payload
+}
+
 func (cpvs CollatorProtocolValidatorSide) handleCollationMessage(
 	sender peer.ID, msg network.NotificationsMessage) (bool, error) {
+
+	// we don't propagate collation messages, so it will always be false
+	propagate := false
+
 	if msg.Type() != network.CollationMsgType {
-		return false, fmt.Errorf("unexpected message type, expected: %d, found:%d",
+		return propagate, fmt.Errorf("%w, expected: %d, found:%d", ErrUnexpectedMessageOnCollationProtocol,
 			network.CollationMsgType, msg.Type())
 	}
 
 	collatorProtocol, ok := msg.(*CollationProtocol)
 	if !ok {
-		return false, fmt.Errorf("failed to cast into collator protocol message, expected: *CollationProtocol, got: %T", msg)
+		return propagate, fmt.Errorf("failed to cast into collator protocol message, expected: *CollationProtocol, got: %T", msg)
 	}
 
 	collatorProtocolV, err := collatorProtocol.Value()
 	if err != nil {
-		return false, fmt.Errorf("getting collator protocol value: %w", err)
+		return propagate, fmt.Errorf("getting collator protocol value: %w", err)
 	}
 	collatorProtocolMessage, ok := collatorProtocolV.(CollatorProtocolMessage)
 	if !ok {
-		return false, errors.New("expected value to be collator protocol message")
+		return propagate, errors.New("expected value to be collator protocol message")
+	}
+
+	collatorProtocolMessageV, err := collatorProtocolMessage.Value()
+	if err != nil {
+		return propagate, fmt.Errorf("getting collator protocol message value: %w", err)
 	}
 
 	switch collatorProtocolMessage.Index() {
-	// TODO: Make sure that V1 and VStaging both types are covered
-	// All the types covered currently are V1.
+	// TODO: Make sure that V1 types are covered.
+	// TODO: Create an issue to cover v2 types.
```

Comment:
is there an issue for this?

Reply:
Just created one https://github.com/ChainSafe/gossamer/issues/3534
Will add more details into it